marin/speedrun/create_leaderboard.py

```diff
@@ -37,28 +54,49 @@ def load_results_file(path: str) -> dict:
 
 
 def create_entry_from_results(results: dict, results_filepath: str) -> LeaderboardEntry:
+    # Path information
     run_name = Path(results_filepath).parent.name
     filepath = Path(results_filepath)
     repo_root = Path(__file__).resolve().parent.parent.parent
     relative_path = filepath.relative_to(repo_root).parent
-    total_training_flops = results["run_stats"]["total_training_flops"]
-    training_time = results["run_stats"]["training_time_in_minutes"]
-    eval_paloma_c4_en_bpb = results["run_stats"]["eval/paloma/c4_en/bpb"]
-    model_size = results["run_related_info"]["num_parameters"]
-    submitted_by = results["run_related_info"].get("submitted_by", "unknown")
-    wandb_link = results["run_related_info"].get("wandb_run_link", None)
-    run_timestamp = results["run_related_info"].get("run_completion_timestamp", None)
+
+    # Get the run info which contains all the information
+    run_data = results["run_info"]
+
+    # Training metrics and FLOPs
+    training_hardware_flops = run_data["training_hardware_flops"]
+    training_time_in_minutes = run_data["training_time_in_minutes"]
+    eval_paloma_c4_en_bpb = run_data["eval/paloma/c4_en/bpb"]
+    model_flops = run_data["model_flops"]
+
+    # Model specification
+    model_size = run_data["model_size"]
+
+    # Run metadata
+    wandb_link = run_data.get("wandb_run_link")
+    run_timestamp = run_data.get("run_completion_timestamp")
+    description = run_data["description"]
+
+    # Author information
+    author_info = run_data["author"]
+    author = SpeedrunAuthor(
+        name=author_info["name"],
+        affiliation=author_info["affiliation"],
+        url=author_info.get("url")
+    )
 
     return LeaderboardEntry(
         run_name=run_name,
-        model_size=model_size,
-        total_training_time=training_time,
-        total_training_flops=total_training_flops,
-        submitted_by=submitted_by,
         results_filepath=str(relative_path),
-        wandb_link=wandb_link,
+        model_size=int(model_size) if model_size is not None else 0,
+        training_hardware_flops=float(training_hardware_flops) if training_hardware_flops is not None else 0.0,
+        training_time_in_minutes=float(training_time_in_minutes) if training_time_in_minutes is not None else 0.0,
```

Comment:
Can we avoid all conversion and just pass it through?  Ideally the whole system would have the same variables (types, names), and if you need to convert, it should be done at the ends, not during serialization.